flux-middle/src/global_env.rs

```diff
@@ -123,14 +131,21 @@ impl<'sess, 'tcx> GlobalEnv<'sess, 'tcx> {
     }
 
     pub fn lookup_fn_sig(&self, def_id: DefId) -> Result<rty::PolySig, UnsupportedFnSig> {
-        match self.fn_sigs.borrow_mut().entry(def_id) {
+        // If it's an extern_fn, resolve it to its local fn_sig's def_id
```

Comment:
This is great! At some point I thought we would need to change everything in the `fhir::Map` to use `DefId` instead of `LocalDefId` but keeping track of just this mapping is brilliant